reading_scraper.py

```diff
@@ -29,115 +53,173 @@ def useragent_generator():
-        if (image == '' and reading.top_image):
-            image = reading.top_image
-    if title == '' or 'Error 404 (Not Found)' in title:
+    try:
+        if reading != 'None' and reading.title != '' and title == '':
+            if ('title' in reading.opengraph):
+                title = reading.opengraph['title'].replace("\\u2019", "'")
+            if (title == '' and reading.title):
+                title = reading.title.replace("\\u2019", "'")
+            
+            if ('description' in reading.opengraph):
+                description = reading.opengraph['description'].replace("\\u2019", "'")
+            if (description == '' and reading.meta_description):
+                description = reading.meta_description.replace("\\u2019", "'")
+            if (len(description) > 500):
+                description = (description[:497] + '...')
+            
+            if ('image' in reading.opengraph):
+                image = reading.opengraph['image']
+            if (image == '' and reading.top_image):
+                image = reading.top_image
+    except:
+        print("Error occurred with getting title: ", sys.exc_info())
+        title = ""
+    if (title == '' 
+        or 'Error 404 (Not Found)' in title
+        or 'Bad Request' in title
+        or title == "Redirecting"
+        or "Page Not Found" in title):
         title = 'Unable to get title of article'
         description = ''
         image = ''
     # print(title)
 
-def get_domain():
+def get_domain(url):
     global domain
+    res = ''
     try:
-        domain = re.search(r'(?:[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?\.)+[a-z0-9][a-z0-9-]{0,61}[a-z0-9]', BASE_URL)
+        domain = re.search(r'(?:[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?\.)+[a-z0-9][a-z0-9-]{0,61}[a-z0-9]', url)
         domain = domain.group()
     except:
         domain = 'Unable to get domain'
+        print("Error occurred with getting domain: ", sys.exc_info())
     # print(domain)
+    res = domain
+    return res
 
 def get_word_count():
     global word_count
-    if reading != 'None':
-        word_list = reading.cleaned_text.split(' ')
-    else:
-        word_list = ['None']
-    word_count = len(word_list)
+    try:
+        if reading != 'None':
+            word_list = reading.cleaned_text.split(' ')
+        else:
+            word_list = ['None']
+        word_count = len(word_list)
+    except:
+        word_count = 0
     # print(word_count)
 
+BASE_URL = transform_url(sys.argv[1])
+
+CACHED = 'https://webcache.googleusercontent.com/search?q=cache:' + BASE_URL
+
+check = check_if_pdf(BASE_URL)
+
+if check:
+    # send to scraper, and let that script create values array
+    sys.exit()
+
 get_reading_data(BASE_URL, CACHED)
 get_title()
-get_domain()
+get_domain(BASE_URL)
 get_word_count()
 
+if ('Unable to get title of article' in title
+    or 'Bloomberg - Are you a robot?' in title
+    or "Access Denied" in title
+    or "Error" in title
```

Comment:
Similar to the "Bloomberg" conditional above, I think this will catch normal articles that have error in them